### DISCUSSION TITLE
NamingConventions change Max Length

### DISCUSSION THREAD

**Discussion by JurjenRoels** [JetBrains/Qodana#271] [Q&A] [answered]
Hello, 

How can I change the Inspection Options in the qodana.yaml file for PhpClassNamingConventionInspection to be Max Lenght 32 instead of default 16.

The link to the available options (GitHub) goes to 404:  https://www.jetbrains.com/help/qodana/2024.2/custom-profiles.html#custom-profiles-examples-inspection-options

**Comment by brichbash** [comment, marked as answer]
Hello,

The option ID for the max length is `m_maxLength`. I think the link in the documentation is not relevant anymore, because profiles in that repo use `yaml` profiles that don't have options there.
Although, it's inconvenient, if you have PhpStorm or a JetBrains IDE, you can edit your inspection profile options there and then go to `./.idea/inspectionProfiles/your-profile.xml` to see the option ID in the changed inspection.
We have [Inspectopedia](https://www.jetbrains.com/help/inspectopedia/PhpClassNamingConventionInspection.html#inspection-options) that lists all IntelliJ inspections, and I think that it could be the place to store option IDs. I'll suggest this idea to the responsible team. It should make editing inspection options in Qodana much easier.

**Reply by JurjenRoels** [reply]
Thank you @brichbash. I think I have a setup now, that works. But I really miss some simple examples of:

- A qodana.yaml using a myqodana.yaml in a directory (profile: - path: myqodana/myqodana.yaml)
- a myqodana.yaml using a custom_inspections in a directory (profile: - path: myqodana/custom_inspections.xml)
-  a custom_inspections.xml that I can update in my IDE.
- How to use code style rules in these configurations and call the docker container.

the myqodana.yaml should show how to exclude certain directories in my repo for instance (vendor/** or **/node_modules/* or explicitly include directories)

I have now created something based upon the qodana_recommended profiles. But I am unsure if it all works the way I want.

I would expect if I update (export) the custom_inspections.xml from the IDE, I would have similar results.

I have raised two items in you track. Where the maxLength is ignored and with phpdoc fully qualified names show up as error.

With regards
Jurjen
